build-system/release-workflows/cut-nightly.js

```diff
@@ -80,94 +97,84 @@ async function getCommit(octokit) {
-  const response = await octokit.rest.git.updateRef({
-    ...params,
-    ref: 'heads/nightly',
-    sha,
-  });
-
-  // Casting to Number because the return type in Octokit is incorrectly
-  // annotated to only ever return 200.
-  switch (Number(response.status)) {
-    case 201:
-      log(
-        'A new',
-        cyan('nightly'),
-        'branch was successfully cut at commit',
-        cyan(sha)
-      );
-      break;
-    case 200:
-      log(
-        'The',
-        cyan('nightly'),
-        'branch is already at the latest',
-        green('green'),
-        'commit',
-        cyan(sha)
-      );
-      break;
-    default:
-      log(
-        red(
-          'An uncaught status was returned while attempting to fast-forward the'
-        ),
-        cyan('nightly'),
-        red('branch to commit'),
-        cyan(sha)
-      );
-      log('See full response:', response);
+  try {
+    await octokit.rest.git.updateRef({
+      ...params,
+      ref: 'heads/nightly',
+      sha,
+    });
+    log(
+      'A new',
+      cyan('nightly'),
+      'branch was successfully cut at commit',
+      cyan(sha)
+    );
+  } catch (e) {
+    log(
+      red(
+        'An uncaught status was returned while attempting to fast-forward the'
+      ),
+      cyan('nightly'),
+      red('branch to commit'),
+      cyan(sha)
+    );
+    log('See full error:', e);
   }
 }
 
 /**
  * Create GitHub tag.
+ *
  * @param {Octokit} octokit
  * @param {string} sha
  * @param {string} ampVersion
  * @return {Promise<void>}
  */
 async function createTag(octokit, sha, ampVersion) {
-  await octokit.rest.git.createTag({
-    ...params,
-    tag: ampVersion,
-    message: ampVersion,
-    object: sha,
-    type: 'commit',
-  });
+  try {
+    await octokit.rest.git.createTag({
+      ...params,
+      tag: ampVersion,
+      message: ampVersion,
+      object: sha,
+      type: 'commit',
+    });
+
+    // once a tag object is created, create a reference.
+    await octokit.rest.git.createRef({
+      ...params,
+      ref: `refs/tags/${ampVersion}`,
+      sha,
+    });
 
-  // once a tag object is created, create a reference
-  const response = await octokit.rest.git.createRef({
-    ...params,
-    ref: `refs/tags/${ampVersion}`,
-    sha,
-  });
-
-  switch (Number(response.status)) {
-    case 201:
-      log(
-        'A new tag',
-        cyan(ampVersion),
-        'was successfully created at commit',
-        cyan(sha)
-      );
-      break;
-    case 422:
+    log(
+      'A new tag',
+      cyan(ampVersion),
+      'was successfully created at commit',
+      cyan(sha)
+    );
+  } catch (e) {
+    if (e instanceof RequestError && e.status === 422) {
       log('The tag', cyan(ampVersion), 'already exists at', cyan(sha));
-      break;
-    default:
+    } else {
       throw new Error(
-        `An unaught status returned while attempting to create a tag\n${response}`
+        `An unaught status returned while attempting to create a tag\n${e}`
```

Comment:
an "unaught" typo!

Reply:
Fixed 60e517e

Reply:
Oops I set this PR to auto-merge so your approval auto-merged. Should I bother with a followup PR on this? :D

Reply:
Nahh